packages/dashboard/src/components/csvDownloadButton/useCSVData.ts

```diff
@@ -0,0 +1,95 @@
+import { Viewport, parseDuration, DataStream, DataPoint } from '@iot-app-kit/core';
+import { useTimeSeriesData, useViewport } from '@iot-app-kit/react-components';
+import { StyledSiteWiseQueryConfig } from '~/customization/widgets/types';
+import { useAssetDescriptionMapQuery } from '~/hooks/useAssetDescriptionQueries';
+import { useQueries } from '~/components/dashboard/queryContext';
+import { CSVDownloadObject } from './types';
+
+const DEFAULT_VIEWPORT = { duration: '10m' };
+
+// Format Date to be a string in format: "YYYY/MM/DD HH:MM:SS"
+export const formatDate = (date: Date) => {
+  const dateString = date.getFullYear().toString() + '/' + (date.getMonth() + 1) + '/' + date.getDate();
+  const timeString = date.toTimeString().split(' ')[0];
+  const dateTime = dateString + ' ' + timeString;
+  return dateTime;
+};
+
+// Check if time is within passed in viewport OR within last x amount of time from request
+const isTimeWithinViewport = (dataPointTimestamp: number, viewport: Viewport, timeOfRequestMS: number) => {
+  const currentPoint = new Date(dataPointTimestamp);
+  if ('duration' in viewport) {
+    // absolute time range
+    const duration = parseDuration(viewport.duration);
+    return currentPoint >= new Date(timeOfRequestMS - duration) && currentPoint <= new Date(timeOfRequestMS);
+  } else {
+    // relative time range
+    return currentPoint >= viewport.start && currentPoint <= viewport.end;
+  }
+};
+
+export const useCSVData = ({
```

Comment:
nit: rename to `useViewportData`? not exactly CSV data but flatten data values during the viewport;
is the plan to wrap this hook under another hook for parsing and downloding?